src/Sylius/Bundle/PayumBundle/spec/Action/ConvertPaymentToPaypalExpressActionSpec.php

```diff
@@ -0,0 +1,118 @@
+use Payum\Core\Request\Convert;
+use PhpSpec\ObjectBehavior;
+use Sylius\Bundle\PayumBundle\Action\ConvertPaymentToPaypalExpressAction;
+use Sylius\Component\Core\Model\AdjustmentInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\PaymentInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+use Sylius\Component\Core\Payment\InvoiceNumberGeneratorInterface;
+use Sylius\Component\Currency\Converter\CurrencyConverterInterface;
+
+/**
+ * @mixin ConvertPaymentToPaypalExpressAction
+ *
+ * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
+ */
+final class ConvertPaymentToPaypalExpressActionSpec extends ObjectBehavior
+{
+    function let(
+        InvoiceNumberGeneratorInterface $invoiceNumberGenerator,
+        CurrencyConverterInterface $currencyConverter
+    ) {
+        $this->beConstructedWith($invoiceNumberGenerator, $currencyConverter);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(ConvertPaymentToPaypalExpressAction::class);
+    }
+
+    function it_implements_action_interface()
+    {
+        $this->shouldImplement(ActionInterface::class);
+    }
+
+    function it_executes_request(
+        InvoiceNumberGeneratorInterface $invoiceNumberGenerator,
+        CurrencyConverterInterface $currencyConverter,
+        Convert $request,
+        PaymentInterface $payment,
+        OrderInterface $order,
+        OrderItemInterface $orderItem,
+        ProductVariantInterface $productVariant,
+        ProductInterface $product
+    ) {
+        $request->getTo()->willReturn('array');
+
+        $payment->getId()->willReturn(19);
+
+        $order->getId()->willReturn(92);
+        $order->getId()->willReturn(92);
+        $order->getCurrencyCode()->willReturn('USD');
+        $order->getTotal()->willReturn(22000);
+        $order->getItems()->willReturn([$orderItem]);
+        $order->getAdjustmentsTotalRecursively(AdjustmentInterface::TAX_ADJUSTMENT)->willReturn(0);
+        $order->getOrderPromotionTotal()->willReturn(0);
+        $order->getShippingTotal()->willReturn(2000);
+
+        $orderItem->getVariant()->willReturn($productVariant);
+        $orderItem->getDiscountedUnitPrice()->willReturn(20000);
+        $orderItem->getQuantity()->willReturn(1);
+
+        $productVariant->getProduct()->willReturn($product);
+
+        $product->getName()->willReturn('Lamborghini Aventador Model');
+
+        $request->getSource()->willReturn($payment);
+        $payment->getOrder()->willReturn($order);
+
+        $invoiceNumberGenerator->generate($order, $payment)->willReturn('19-92');
+        $currencyConverter->convertFromBase(22000, 'USD')->willReturn(22000);
+        $currencyConverter->convertFromBase(20000, 'USD')->willReturn(20000);
+        $currencyConverter->convertFromBase(2000, 'USD')->willReturn(2000);
```

Comment:
Please return some other number than the original ones. Right now we have no information if `currencyConverter` was even used.